src/contrib/cluster/Akka.Cluster.Tools/PublishSubscribe/DistributedMessages.cs

```diff
@@ -415,6 +415,22 @@ public override string ToString()
             return $"Publish<topic:{Topic}, sendOneToEachGroup:{SendOneMessageToEachGroup}, message:{Message}>";
         }
     }
+    
+    public sealed record PublishWithAck(
+        string Topic,
+        object Message,
+        TimeSpan Timeout,
```

Comment:
I'd add a CTOR implementation just to quickly validate the timeout or you can do validation inside the mediator and send a failure message back. Probably better to throw an exception inside the publisher.